src/obstacle_infos_publisher.cpp

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <Eigen/Dense>
+#include <boost/shared_ptr.hpp>
+#include <fstream>
+#include <memory>
+#include <vector>
+
+#include <pinocchio/fwd.hpp>
+#include <pinocchio/multibody/model.hpp>
+#include <yaml-cpp/yaml.h>
+#include <ros/ros.h>
+
+namespace panda_torque_mpc {
+
+class ObstacleInfosPublisher {
+  ObstacleInfosPublisher(
+      const std::shared_ptr<ros::NodeHandle> &ph,
+      const boost::shared_ptr<pinocchio::Model> &pin_model_,
+      const boost::shared_ptr<pinocchio::GeometryModel> &collision_model_);
+  ~ObstacleInfosPublisher();
+  {
+    // Loading the Yaml file
+    YAML::Node config = YAML::LoadFile("config.yaml");
+
+    const YAML::Node& obstacles = config["obstacles"];
+    // Going through all the obstacles and adding them to the geom model
+    for (const auto& obstacle : obstacles)
+    {
+
+    }
```

Comment:
You should avoid parsing the YAML directly as much as you can.
You can transform the YAML to a form:
```yaml
obstacle1:
    type: "sphere"
    radious: 0.175
obstacle2:
    type: "box"
    dim: [1, 1, 1]
```

This way you can parse it with following pseudocode:
```
col_idx = 0
obstacles = []
while nh->has_param("obstacle_" + col_idx)
    obstacle = "obstacle_" + col_idx
    type = nh->get_param(obstacle + "/type")
    if type is "sphere"
        radious = nh->get_param(obstacle + "/radious")
        obstacle = Spehere(radious)
    else if type is "box"
        dim = nh->get_param(obstacle + "/dim")
        obstacle = Box(dim[0], dim[1], dim[2])
    end
    obstacles.append(obstacle)
    col_idx += 1
end
```